### PR TITLE
Add file type component

### PR DESCRIPTION
**Pull request by marcelmanz**
I added this to my personal config and I think it gives a good touch, but it has the icons dependency. Tell me what you think and I finish it.

I did not check what needs to be done yet to implement the component correctly.

### CODE REVIEW

File: lua/sttusline/components/file-type.lua [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+		"BufWritePost",
+		"BufReadPost",
+	},
+	configs = {
+		style = "default",
+	},
+	padding = 2,
+	user_event = "VeryLazy",
+	update = function()
+		local icon = require("nvim-web-devicons").get_icon(
```
Comment (marcelmanz) [inline lua/sttusline/components/file-type.lua]: an alternative in the case that `nvim-web-devicons` is not available is to use:

```lua
update = function()
    local has_devicons, web_devicons = pcall(require, "nvim-web-devicons")
    local ft

    if has_devicons then
        ft = web_devicons.get_icon(
            vim.fn.expand "%:t",
            vim.fn.expand "%:e",
            { default = true }
        )
    else
        ft = vim.bo.filetype
    end

    return string.format("%s", ft)
end,
```

Comment (sontungexpt) [reply lua/sttusline/components/file-type.lua]: Yah, feel free to add it if needed.

Comment (marcelmanz) [reply lua/sttusline/components/file-type.lua]: Done in 7cfaecd

Comment (marcelmanz) [reply lua/sttusline/components/file-type.lua]: Is there anything else needed to merge this pr @sontungexpt ?

**Comment by sontungexpt** [comment]
Thanks for PR

**Comment by sontungexpt** [comment]
Apologies for the revert, but I've noticed that the icon only has a single color. Could you please add more colors to it? It will be more beautiful.

**Pull request merged**